Data/models/DataSaved.swift

```diff
@@ -0,0 +1,52 @@
+// Copyright 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import CoreData
+
+public final class DataSaved: NSManagedObject, CRUD {
+    @NSManaged public var savedUrl: String
+    @NSManaged public var amount: String
+    
+    public class func get(with savedUrl: String) -> DataSaved? {
+        let predicate = NSPredicate(format: "\(#keyPath(DataSaved.savedUrl)) == %@", savedUrl)
+        return first(where: predicate)
+    }
+    
+    public class func all() -> [DataSaved] {
+        all() ?? []
+    }
+    
+    public class func delete(with savedUrl: String) {
+        let context = DataController.viewContext
+        
+        if let item = get(with: savedUrl) {
+            item.delete(context: .existing(context))
```

Comment:
Do we have to delete on main context?
In general write operations should be done using background contexts/threads

we do it like this:
````
DataController.perform { context in 
   // use the context from the closure
}
```
You can see in other Data classes how we usually do it.

Reply:
Add/Delete operations are moved to background thread.